docs/rfcs/1011-non-blocking-invite-transport.md

```diff
@@ -0,0 +1,1238 @@
+                    },
+                    {
+                        "name": "reason",
+                        "type": "CancelledAttemptReason"
+                    }
+                ]
+            }
+        ]
+    }
+]
+```
+
+
+#### `invite_greeter_step` command
+
+This authenticated command, along with `invite_claimer_step`, is the main route used to establish a secure channel between both peers.
+
+It is used by the greeter to submit the data for a given step. The step is identified by:
+- the author of the command (being the greeter)
+- the invitation token
+- the type of the submitted step data
+
+For the step data to be accepted, it has to pass the following the checks:
+- the invitation exists
+- the invitation is pending (i.e. not completed nor cancelled)
+- the author is allowed (i.e. it is part of the greeters)
+- the attempt exists
+- the attempt is active (i.e. joined and not cancelled)
+- the previous steps must have been completed by both peers
+
+Once the data is accepted, it either returns:
+- the claimer step data if it is available
+- a `not_ready` status otherwise
+
+In the second case, the author can poll the command with the same parameters until it gets the claimer step data.
+
+If the submitted step differs from the one that has already been registered, a dedicated `step_mismatch` status is returned.
+
+Pseudo-code:
+
+```python
+def invite_greeter_step(
+    token: InvitationToken,
+    attempt: AttemptID,
+    step: GreeterStep,
+) ->
+  InvitationNotFound | InvitationCompleted | InvitationCancelled
+  GreeterNotAllowed
+  AttemptNotFound | AttemptNotJoined | AttemptCancelled
+  StepTooAdvanced | PayloadMismatch
+  NotReady | OK[ClaimerStep]:
+    # Perform checks
+    [...]
+    # Call `set_greeter_payload`
+    [...]
+    # Return `greeter_step` or `NotReady`
+    [...]
+```
+
+Schema definition:
+
+```json5
+[
+    {
+        "major_versions": [
+            4
+        ],
+        "req": {
+            "cmd": "invite_greeter_step",
+            "fields": [
+                {
+                    "name": "token",
+                    "type": "InvitationToken"
+                },
+                {
+                    "name": "attempt",
+                    "type": "AttemptID"
+                },
+                {
+                    "name": "greeter_step",
+                    "type": "GreeterStep"
+                }
+            ]
+        },
+        "reps": [
+            {
+                "status": "ok",
+                "fields": [
+                    {
+                        "name": "claimer_step",
+                        "type": "ClaimerStep"
+                    }
+                ]
+            },
+            {
+                // The claimer has not submitted its step yet
+                "status": "not_ready"
+            },
+            {
+                // The invitation token doesn't correspond to any existing invitation
+                "status": "invitation_not_found"
```

Comment:
remove and only rely on `attempt_not_found`

Reply:
ccf6d568d950206e8b632bc2469462e7910f790c